### ISSUE TITLE
`isoweek_to_datetime` not working with pl.DataFrame in a `with_columns` context

### ISSUE THREAD

**Issue by pietroppeter** [FBruzzesi/iso-week-date#83]
I am trying to use `isoweek_to_datetime` in a dataframe to generate a new column, it seems the function is not working in a `with_columns` context.

Reproducible example (adapted from [docs](https://fbruzzesi.github.io/iso-week-date/api/polars/#iso_week_date.polars_utils.SeriesIsoWeek.isoweek_to_datetime) and generated by exporting a jupyter notebook to markdown):

```python
import polars as pl
from iso_week_date.polars_utils import SeriesIsoWeek  # noqa: F401

s = pl.Series(["2022-W52", "2023-W01", "2023-W02"])
s.iwd.isoweek_to_datetime()
```

<pre style="white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace"></pre>

<div><style>
.dataframe > thead > tr,
.dataframe > tbody > tr {
  text-align: right;
  white-space: pre-wrap;
}
</style>
<small>shape: (3,)</small><table border="1" class="dataframe"><thead><tr><th></th></tr><tr><td>date</td></tr></thead><tbody><tr><td>2022-12-26</td></tr><tr><td>2023-01-02</td></tr><tr><td>2023-01-09</td></tr></tbody></table></div>

```python
df = pl.DataFrame(data=dict(isoweek=s))
df
```

<pre style="white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace"></pre>

<div><style>
.dataframe > thead > tr,
.dataframe > tbody > tr {
  text-align: right;
  white-space: pre-wrap;
}
</style>
<small>shape: (3, 1)</small><table border="1" class="dataframe"><thead><tr><th>isoweek</th></tr><tr><td>str</td></tr></thead><tbody><tr><td>&quot;2022-W52&quot;</td></tr><tr><td>&quot;2023-W01&quot;</td></tr><tr><td>&quot;2023-W02&quot;</td></tr></tbody></table></div>

```python
df.get_column("isoweek").iwd.isoweek_to_datetime().alias("week_date")
```

<pre style="white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace"></pre>

<div><style>
.dataframe > thead > tr,
.dataframe > tbody > tr {
  text-align: right;
  white-space: pre-wrap;
}
</style>
<small>shape: (3,)</small><table border="1" class="dataframe"><thead><tr><th>week_date</th></tr><tr><td>date</td></tr></thead><tbody><tr><td>2022-12-26</td></tr><tr><td>2023-01-02</td></tr><tr><td>2023-01-09</td></tr></tbody></table></div>

```python
df.with_columns(pl.col("isoweek").iwd.isoweek_to_datetime().alias("week_date"))
```

<pre style="white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace"><span style="color: #800000; text-decoration-color: #800000">╭─────────────────────────────── </span><span style="color: #800000; text-decoration-color: #800000; font-weight: bold">Traceback </span><span style="color: #bf7f7f; text-decoration-color: #bf7f7f; font-weight: bold">(most recent call last)</span><span style="color: #800000; text-decoration-color: #800000"> ────────────────────────────────╮</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> in <span style="color: #00ff00; text-decoration-color: #00ff00">&lt;module&gt;</span>:<span style="color: #0000ff; text-decoration-color: #0000ff">1</span>                                                                                    <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>                                                                                                  <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #800000; text-decoration-color: #800000">❱ </span>1 df.with_columns(pl.col(<span style="color: #808000; text-decoration-color: #808000">"isoweek"</span>).iwd.isoweek_to_datetime().alias(<span style="color: #808000; text-decoration-color: #808000">"week_date"</span>))              <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">2 </span>                                                                                             <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>                                                                                                  <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #bfbf7f; text-decoration-color: #bfbf7f">/Users/pietropeterlongo/projects/poste-logistics/.venv/lib/python3.12/site-packages/iso_week_dat</span> <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #bfbf7f; text-decoration-color: #bfbf7f">e/</span><span style="color: #808000; text-decoration-color: #808000; font-weight: bold">polars_utils.py</span>:<span style="color: #0000ff; text-decoration-color: #0000ff">453</span> in <span style="color: #00ff00; text-decoration-color: #00ff00">isoweek_to_datetime</span>                                                     <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>                                                                                                  <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">450 </span><span style="color: #bfbf7f; text-decoration-color: #bfbf7f">│   │   </span><span style="color: #808000; text-decoration-color: #808000">'''</span>                                                                                <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">451 </span><span style="color: #bfbf7f; text-decoration-color: #bfbf7f">│   │   </span><span style="color: #808000; text-decoration-color: #808000">```</span>                                                                                <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">452 </span><span style="color: #bfbf7f; text-decoration-color: #bfbf7f">│   │   </span><span style="color: #808000; text-decoration-color: #808000">"""</span>                                                                                <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #800000; text-decoration-color: #800000">❱ </span>453 <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   </span><span style="color: #0000ff; text-decoration-color: #0000ff">return</span> isoweek_to_datetime(<span style="color: #00ffff; text-decoration-color: #00ffff">self</span>._series, offset=offset, weekday=weekday)           <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">454 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   </span>                                                                                       <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">455 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   </span><span style="color: #0000ff; text-decoration-color: #0000ff">def</span> <span style="color: #00ff00; text-decoration-color: #00ff00">isoweekdate_to_datetime</span>(<span style="color: #00ffff; text-decoration-color: #00ffff">self</span>: Self, offset: OffsetType = timedelta(<span style="color: #0000ff; text-decoration-color: #0000ff">0</span>)) -&gt; T:       <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">456 </span><span style="color: #bfbfbf; text-decoration-color: #bfbfbf">│   │   </span><span style="color: #808000; text-decoration-color: #808000">"""Converts `str` series or expr of ISO Week date format YYYY-WNN-D to a series </span>   <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>                                                                                                  <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #bfbf7f; text-decoration-color: #bfbf7f">/Users/pietropeterlongo/projects/poste-logistics/.venv/lib/python3.12/site-packages/iso_week_dat</span> <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #bfbf7f; text-decoration-color: #bfbf7f">e/</span><span style="color: #808000; text-decoration-color: #808000; font-weight: bold">polars_utils.py</span>:<span style="color: #0000ff; text-decoration-color: #0000ff">186</span> in <span style="color: #00ff00; text-decoration-color: #00ff00">isoweek_to_datetime</span>                                                     <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>                                                                                                  <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">183 </span><span style="color: #bfbf7f; text-decoration-color: #bfbf7f">│   </span><span style="color: #808000; text-decoration-color: #808000">'''</span>                                                                                    <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">184 </span><span style="color: #bfbf7f; text-decoration-color: #bfbf7f">│   </span><span style="color: #808000; text-decoration-color: #808000">```</span>                                                                                    <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">185 </span><span style="color: #bfbf7f; text-decoration-color: #bfbf7f">│   </span><span style="color: #808000; text-decoration-color: #808000">"""</span>                                                                                    <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #800000; text-decoration-color: #800000">❱ </span>186 <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   </span><span style="color: #0000ff; text-decoration-color: #0000ff">if</span> <span style="color: #ff00ff; text-decoration-color: #ff00ff">not</span> is_isoweek_series(series):                                                      <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">187 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   </span>msg = <span style="color: #808000; text-decoration-color: #808000">f"`series` values must match ISO Week format {</span>ISOWEEK__FORMAT<span style="color: #808000; text-decoration-color: #808000">}"</span>              <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">188 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   </span><span style="color: #0000ff; text-decoration-color: #0000ff">raise</span> <span style="color: #00ffff; text-decoration-color: #00ffff">ValueError</span>(msg)                                                              <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">189 </span>                                                                                           <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>                                                                                                  <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #bfbf7f; text-decoration-color: #bfbf7f">/Users/pietropeterlongo/projects/poste-logistics/.venv/lib/python3.12/site-packages/polars/expr/</span> <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #808000; text-decoration-color: #808000; font-weight: bold">expr.py</span>:<span style="color: #0000ff; text-decoration-color: #0000ff">152</span> in <span style="color: #00ff00; text-decoration-color: #00ff00">__bool__</span>                                                                          <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>                                                                                                  <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">  149 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   │   </span><span style="color: #808000; text-decoration-color: #808000">"- instead of `pl.col('a') in [y, z]`, use `pl.col('a').is_in([y, z])`\n"</span>    <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">  150 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   │   </span><span style="color: #808000; text-decoration-color: #808000">"- instead of `max(pl.col('a'), pl.col('b'))`, use `pl.max_horizontal(pl.col</span> <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">  151 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   </span>)                                                                                <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #800000; text-decoration-color: #800000">❱ </span>  152 <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   </span><span style="color: #0000ff; text-decoration-color: #0000ff">raise</span> <span style="color: #00ffff; text-decoration-color: #00ffff">TypeError</span>(msg)                                                             <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">  153 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   </span>                                                                                     <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">  154 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   </span><span style="color: #0000ff; text-decoration-color: #0000ff">def</span> <span style="color: #00ff00; text-decoration-color: #00ff00">__abs__</span>(<span style="color: #00ffff; text-decoration-color: #00ffff">self</span>) -&gt; Expr:                                                           <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">  155 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   </span><span style="color: #0000ff; text-decoration-color: #0000ff">return</span> <span style="color: #00ffff; text-decoration-color: #00ffff">self</span>.abs()                                                                <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">╰──────────────────────────────────────────────────────────────────────────────────────────────────╯</span>
<span style="color: #ff0000; text-decoration-color: #ff0000; font-weight: bold">TypeError: </span>the truth value of an Expr is ambiguous

You probably got here by using a Python standard library function instead of the native expressions API.
Here are some things you might want to try:
- instead of `<span style="color: #800080; text-decoration-color: #800080; font-weight: bold">pl.col</span><span style="font-weight: bold">(</span><span style="color: #008000; text-decoration-color: #008000">'a'</span><span style="font-weight: bold">)</span> and <span style="color: #800080; text-decoration-color: #800080; font-weight: bold">pl.col</span><span style="font-weight: bold">(</span><span style="color: #008000; text-decoration-color: #008000">'b'</span><span style="font-weight: bold">)</span>`, use `<span style="color: #800080; text-decoration-color: #800080; font-weight: bold">pl.col</span><span style="font-weight: bold">(</span><span style="color: #008000; text-decoration-color: #008000">'a'</span><span style="font-weight: bold">)</span> &amp; <span style="color: #800080; text-decoration-color: #800080; font-weight: bold">pl.col</span><span style="font-weight: bold">(</span><span style="color: #008000; text-decoration-color: #008000">'b'</span><span style="font-weight: bold">)</span>`
- instead of `<span style="color: #800080; text-decoration-color: #800080; font-weight: bold">pl.col</span><span style="font-weight: bold">(</span><span style="color: #008000; text-decoration-color: #008000">'a'</span><span style="font-weight: bold">)</span> in <span style="font-weight: bold">[</span>y, z<span style="font-weight: bold">]</span>`, use `<span style="color: #800080; text-decoration-color: #800080; font-weight: bold">pl.col</span><span style="font-weight: bold">(</span><span style="color: #008000; text-decoration-color: #008000">'a'</span><span style="font-weight: bold">)</span><span style="color: #800080; text-decoration-color: #800080; font-weight: bold">.is_in</span><span style="font-weight: bold">([</span>y, z<span style="font-weight: bold">])</span>`
- instead of `<span style="color: #800080; text-decoration-color: #800080; font-weight: bold">max</span><span style="font-weight: bold">(</span><span style="color: #800080; text-decoration-color: #800080; font-weight: bold">pl.col</span><span style="font-weight: bold">(</span><span style="color: #008000; text-decoration-color: #008000">'a'</span><span style="font-weight: bold">)</span>, <span style="color: #800080; text-decoration-color: #800080; font-weight: bold">pl.col</span><span style="font-weight: bold">(</span><span style="color: #008000; text-decoration-color: #008000">'b'</span><span style="font-weight: bold">))</span>`, use `<span style="color: #800080; text-decoration-color: #800080; font-weight: bold">pl.max_horizontal</span><span style="font-weight: bold">(</span><span style="color: #800080; text-decoration-color: #800080; font-weight: bold">pl.col</span><span style="font-weight: bold">(</span><span style="color: #008000; text-decoration-color: #008000">'a'</span><span style="font-weight: bold">)</span>, <span style="color: #800080; text-decoration-color: #800080; font-weight: bold">pl.col</span><span style="font-weight: bold">(</span><span style="color: #008000; text-decoration-color: #008000">'b'</span><span style="font-weight: bold">))</span>`

</pre>

workaround:

```python
df["week_date"] = df.get_column("isoweek").iwd.isoweek_to_datetime().alias("week_date")
df
```

<pre style="white-space:pre;overflow-x:auto;line-height:normal;font-family:Menlo,'DejaVu Sans Mono',consolas,'Courier New',monospace"><span style="color: #800000; text-decoration-color: #800000">╭─────────────────────────────── </span><span style="color: #800000; text-decoration-color: #800000; font-weight: bold">Traceback </span><span style="color: #bf7f7f; text-decoration-color: #bf7f7f; font-weight: bold">(most recent call last)</span><span style="color: #800000; text-decoration-color: #800000"> ────────────────────────────────╮</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> in <span style="color: #00ff00; text-decoration-color: #00ff00">&lt;module&gt;</span>:<span style="color: #0000ff; text-decoration-color: #0000ff">1</span>                                                                                    <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>                                                                                                  <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #800000; text-decoration-color: #800000">❱ </span>1 df[<span style="color: #808000; text-decoration-color: #808000">"week_date"</span>] = df.get_column(<span style="color: #808000; text-decoration-color: #808000">"isoweek"</span>).iwd.isoweek_to_datetime().alias(<span style="color: #808000; text-decoration-color: #808000">"week_date"</span>)      <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">2 </span>df                                                                                           <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">3 </span>                                                                                             <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>                                                                                                  <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #bfbf7f; text-decoration-color: #bfbf7f">/Users/pietropeterlongo/projects/poste-logistics/.venv/lib/python3.12/site-packages/polars/dataf</span> <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #bfbf7f; text-decoration-color: #bfbf7f">rame/</span><span style="color: #808000; text-decoration-color: #808000; font-weight: bold">frame.py</span>:<span style="color: #0000ff; text-decoration-color: #0000ff">1211</span> in <span style="color: #00ff00; text-decoration-color: #00ff00">__setitem__</span>                                                                <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>                                                                                                  <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f"> 1208 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   │   │   </span><span style="color: #808000; text-decoration-color: #808000">"DataFrame object does not support `Series` assignment by index"</span>         <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f"> 1209 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   │   │   </span><span style="color: #808000; text-decoration-color: #808000">"\n\nUse `DataFrame.with_columns`."</span>                                      <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f"> 1210 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   │   </span>)                                                                            <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span> <span style="color: #800000; text-decoration-color: #800000">❱ </span> 1211 <span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   │   </span><span style="color: #0000ff; text-decoration-color: #0000ff">raise</span> <span style="color: #00ffff; text-decoration-color: #00ffff">TypeError</span>(msg)                                                         <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f"> 1212 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   </span>                                                                                 <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f"> 1213 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f"># df[["C", "D"]]</span>                                                                 <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">│</span>   <span style="color: #7f7f7f; text-decoration-color: #7f7f7f"> 1214 </span><span style="color: #7f7f7f; text-decoration-color: #7f7f7f">│   │   </span><span style="color: #0000ff; text-decoration-color: #0000ff">elif</span> <span style="color: #00ffff; text-decoration-color: #00ffff">isinstance</span>(key, <span style="color: #00ffff; text-decoration-color: #00ffff">list</span>):                                                      <span style="color: #800000; text-decoration-color: #800000">│</span>
<span style="color: #800000; text-decoration-color: #800000">╰──────────────────────────────────────────────────────────────────────────────────────────────────╯</span>
<span style="color: #ff0000; text-decoration-color: #ff0000; font-weight: bold">TypeError: </span>DataFrame object does not support `Series` assignment by index

Use `DataFrame.with_columns`.
</pre>

(note that the workaround does not work)

**Comment by pietroppeter** [comment]
ah I found the workaround that works:
`df.with_columns(df.get_column("isoweek").iwd.isoweek_to_datetime().alias("week_date"))`

**Comment by FBruzzesi** [comment]
That's certainly a bug. Everything that works with series' should work with expr's as well.

I wonder if the [extensions registration](https://github.com/FBruzzesi/iso-week-date/blob/main/iso_week_date/polars_utils.py#L328) ends in some kind of conflict:

```py
@pl.api.register_series_namespace("iwd")
@pl.api.register_expr_namespace("iwd")
class SeriesIsoWeek(Generic[T]):
    ...
```

I will investigate this further.

Could you share the polars version you are using?

**Comment by FBruzzesi** [comment]
Ignore all previous considerations. The error 

> TypeError: the truth value of an Expr is ambiguous

is triggered in the line:

```py
return series.str.extract(pattern).is_not_null().all()
```

which for Expr seems to cause issues. I will think how to deal with this (most likely with a duck-typing approach)

**Comment by FBruzzesi** [comment]
@pietroppeter should be fixed in `v1.4.0`

**Comment by pietroppeter** [comment]
Will check and report back
